### PR TITLE
Use ambient Nix when available, with a two stage fallback

### LINKED ISSUE #451: nix-direnv should not bundle nix
Re: https://github.com/NixOS/nixpkgs/issues/270741

The nix-direnv derivation wraps `direnvrc` by prepending the file with `NIX_BIN_PREFIX=${pkgs.nix}`.

As a result:
- On non-NixOS systems it's impossible to use nix-direnv with the system version of nix. This is very surprising and causes a number of issues like warnings for unsupported Nix options.
- On NixOS it doesn't take into account `config.nix.package`, instead opting for the hardcoded `pkgs.nix`

In my opinion it doesn't make sense to bundle nix with nix-direnv. I think it's reasonable to expect nix to be in `PATH` on any system where people would actually want to use nix-direnv.

### PR DESCRIPTION
**Pull request by K900**
First, we try to use the ambient Nix version.
Then, we try to use $NIX_DIRENV_FALLBACK_NIX, which is set by default, but can also be overridden by the user.
Only then, if neither is available, we fail.

Fixes #451.

### CODE REVIEW

**Review by bbenne10 — APPROVED** [review]
This looks great to me. I appreciate the effort. 

I will wait for @Mic92 to have a chance to review before I merge (though they may get around to it first).

File: default.nix
Comment (Mic92) [inline default.nix:61]: Can you mention this environment variable in the README? Otherwise it looks good to me.

Comment (K900) [reply default.nix:61]: Done.

**Comment by Mic92** [comment]
@mergify queue

**Comment by K900** [comment]
Is it just broken now...?

**Comment by Mic92** [comment]
> Is it just broken now...?

"required checks" in mergify.yml needs an update.

**Pull request merged**
